impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java

```diff
@@ -210,92 +215,95 @@ public Stream<ConfigHolder> getBrokerConfigs(Supplier<KafkaEndpoints> endPointCo
      */
     @NotNull
     public ConfigHolder generateConfigForSpecificNode(KafkaEndpoints kafkaEndpoints, int nodeId) {
-        Properties server = new Properties();
-        server.putAll(brokerConfigs);
+        final var role = determineRole(nodeId);
+        Properties nodeConfiguration = new Properties();
+        nodeConfiguration.putAll(brokerConfigs);
 
-        putConfig(server, "broker.id", Integer.toString(nodeId));
-
-        var interBrokerEndpoint = kafkaEndpoints.getEndpointPair(Listener.INTERNAL, nodeId);
-        var clientEndpoint = kafkaEndpoints.getEndpointPair(Listener.EXTERNAL, nodeId);
-        var anonEndpoint = kafkaEndpoints.getEndpointPair(Listener.ANON, nodeId);
-
-        // - EXTERNAL: used for communications to/from consumers/producers optionally with authentication
-        // - ANON: used for communications to/from consumers/producers without authentication primarily for the extension to validate the cluster
-        // - INTERNAL: used for inter-broker communications (always no auth)
-        // - CONTROLLER: used for inter-broker controller communications (kraft - always no auth)
-
-        var externalListenerTransport = securityProtocol == null ? SecurityProtocol.PLAINTEXT.name() : securityProtocol;
+        putConfig(nodeConfiguration, "broker.id", Integer.toString(nodeId));
 
         var protocolMap = new TreeMap<String, String>();
         var listeners = new TreeMap<String, String>();
         var advertisedListeners = new TreeMap<String, String>();
         var earlyStart = new TreeSet<String>();
 
-        protocolMap.put("EXTERNAL", externalListenerTransport);
-        listeners.put("EXTERNAL", clientEndpoint.listenAddress());
-        advertisedListeners.put("EXTERNAL", clientEndpoint.advertisedAddress());
-
-        protocolMap.put("ANON", SecurityProtocol.PLAINTEXT.name());
-        listeners.put("ANON", anonEndpoint.listenAddress());
-        advertisedListeners.put("ANON", anonEndpoint.advertisedAddress());
-
-        protocolMap.put("INTERNAL", SecurityProtocol.PLAINTEXT.name());
-        listeners.put("INTERNAL", interBrokerEndpoint.listenAddress());
-        advertisedListeners.put("INTERNAL", interBrokerEndpoint.advertisedAddress());
-        earlyStart.add("INTERNAL");
-        putConfig(server, "inter.broker.listener.name", "INTERNAL");
+        final ConfigHolder configHolder;
+        if (role.contains(BROKER_ROLE)) {
```

Comment:
Worth pulling out a `brokerConfigHolder(...)` method and a `controllerOnlyConfigHolder(...)` type method? This block is big.